site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+# GEP-709: Cross Namespace References from Routes
+
+* Issue URL: https://github.com/kubernetes-sigs/gateway-api/issues/709
+* Status: Implementable
+
+## TLDR
+
+This GEP attempts to tackle both cross namespace forwarding and route inclusion.
+These are closely related concepts that can be solved with a new ReferencePolicy
+resource that enables app admins to describe where they trust references from.
+
+## Motivation/User Journeys/Background
+
+This GEP enables the following capabilities:
+
+1. Retaining full control of Gateway and Routes in an infra namespace, while
+   targeting apps in different namespaces.
+1. Traffic splitting between Services in different namespaces.
+1. Mesh overrides to target Services in different namespaces. 
+
+## ReferencePolicy
+
+Anytime we allow crossing a namespace boundary, we need to be very cautious. To
+ensure that this feature is safe, we need to enforce a handshake mechanism that
+requires resources in both namespaces to agree to this reference. To accomplish
+that, a new ReferencePolicy resource should be introduced.
+
+![Reference Policy](images/709-referencepolicy.png)
+
+With this model, Routes would be able to directly reference Routes and Services
+in other namespaces. These references would only be considered valid if a
+ReferencePolicy in the target namespace explicitly allowed it. 
+
+The following example shows how a HTTPRoute in namespace foo could reference
+a Service in namespace bar. In this example a ReferencePolicy in the bar
+namespace explicitly allows references to Services from HTTPRoutes in the foo
+namespace.
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  rules:
+  - matches:
+    - path: /bar
+    forwardTo:
+      backend:
+      - name: bar
+        namespace: bar
+---
+kind: ReferencePolicy
```

Comment:
We might want to call this "InboundReferencePolicy" or something make the direction obvious

Reply:
That's a good idea. Any thoughts on this name change @youngnick @hbagdi @jpeach @howardjohn?

Reply:
I'm okay with it, although it's a little long. I had discounted it before because I don't see us adding an `OutboundReferencePolicy` at any point, so I didn't think the disambiguation was necessary. But I don't object if others think it makes things clearer, I'm a little deep into this design to be able to have a good idea of how the name looks at first glance.

Reply:
I think ReferencePolicy is find. (per discussion)